homeassistant/components/spotify/__init__.py

```diff
@@ -53,39 +54,38 @@ async def async_setup_entry(hass: HomeAssistant, entry: SpotifyConfigEntry) -> b
     except aiohttp.ClientError as err:
         raise ConfigEntryNotReady from err
 
-    spotify = Spotify(auth=session.token["access_token"])
+    spotify = SpotifyClient(async_get_clientsession(hass))
 
-    coordinator = SpotifyCoordinator(hass, spotify, session)
+    spotify.authenticate(session.token[CONF_ACCESS_TOKEN])
 
-    await coordinator.async_config_entry_first_refresh()
+    refresh_lock = asyncio.Lock()
 
-    async def _update_devices() -> list[dict[str, Any]]:
-        if not session.valid_token:
+    async def _refresh_token() -> str:
+        async with refresh_lock:
```

Comment:
`async_ensure_token_valid` already has a lock, you added it yourself @joostlek in https://github.com/home-assistant/core/pull/117912. Why do we need another lock?

Reply:
I actually don't know lol, when I made this I did add it with a reason, but I can't remember which one it was now